casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -810,7 +810,9 @@ object BlockAPI {
   }
 
   def exploratoryDeploy[F[_]: Monad: EngineCell: Log: SafetyOracle: BlockStore](
-      term: String
+      term: String,
+      blockHash: String,
+      isPostStateHash: Boolean
```

Comment:
Whats the reason behind this flag? Is it to compute something on state of merged parents for debug purposes?

Reply:
There are `postStateHash` and `preStateHash` in a block. If the isPostStateHash is `false`, it would choose `preStateHash`. That's  my idea.

Reply:
Maybe give another name for the flag which makes more sense. Any suggestions?

Reply:
I suggest make it `usePreStateHash`, with default value `false`, as general purpose of this method is to evaluate state of particular block, so of `postStateHash`. I can see `usePreStateHash=true` for debug purposes, as in general this state does not represent anything particular.